WIP on running local against test tenant

yuzefovich opened the pull request:
Epic: None

yuzefovich committed:
WIP on running local against test tenant

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107300)
<!-- Reviewable:end -->

yuzefovich committed:
randgen: skip virtual columns in generate_test_objects

If we don't, then `validateTableIndexes` will fail. Found when running
`local` logic tests against the test tenant.

Epic: None

Release note: None
